include/xlnt/utils/date.hpp

```diff
@@ -61,8 +68,9 @@ struct XLNT_API date
     /// <summary>
     /// Calculates and returns the day of the week that this date represents in the range
     /// 0 to 6 where 0 represents Sunday.
+    /// Returns an empty optional if the weekday could not be determined.
     /// </summary>
-    int weekday() const;
+    optional<int> weekday() const;
```

Comment:
This changes a public interface. I don't think we should to this with a minor release. I would just return -1 in case of a failure, at least for now.

Reply:
While I really dislike magic numbers like -1 (often used in old C/C++ functions) as they are often forgotten to be checked, and while I also don't think that `xlnt::date::weekday` and `xlnt::datetime::weekday` are used that widely (or at least so much that a change would be a lot of work), I do agree that releasing a new version of XLNT that only contains bug fixes is more important right now. Thus, I reverted my change and documented that -1 will be returned.

Reply:
However, I would highly prefer to change it to an `xlnt::optional` instead of using magic numbers like -1 for the next major release.